app/auth/controllers.py

```diff
@@ -211,9 +211,13 @@ def get_s3_signed_url():
             return handle_error('INVALID_INPUT',
                                 'md5 hash can not be empty',
                                 400)
+        if path == 'datapackage.json':
+            return handle_error('INVALID_INPUT',
+                                'datapackage.json should not publish with this api',
```

Comment:
I think the error message could be more descriptive e.g. "You cannot direct upload datapackage.json to the BitStore but must use the publishing API."